azext_iot/common/certops.py

```diff
@@ -102,6 +102,52 @@ def create_self_signed_certificate(
     return result
 
 
+def isBase64(content: str) -> bool:
+    """
+    Checks if certificant content should be valid base64 string without prefix and suffix
+
+    Args:
+        content (str): certificant content without prefix and suffix.
+
+    Returns:
+        isBase64 (bool): returns where the certificant content is valid base64 value.
+    """
+    try:
+        sb_bytes = bytes(content, "ascii")
+        base64.b64decode(sb_bytes)
+    except Exception:
+        return False
+    return True
+
+
+def getCertificateFormatValidation(certificate: str) -> str:
+    """
+    Checks if the certificate format is valid
+    1. contain matched BEGIN and END segments
+    2. content should be valid base64 string without BEGIN and END segments
+
+    Args:
+        certificate (str): certificate string.
+
+    Returns:
+        validation string (str): returns validation string when content format is incorrect.
+    """
+    if certificate.find("CERTIFICATE") != -1:
+        if (
+            certificate.find("-----BEGIN CERTIFICATE-----") != -1
+            and certificate.find("-----END CERTIFICATE-----") != -1
+        ):
+            certificate = certificate.replace("-----BEGIN CERTIFICATE-----", "")
+            certificate = certificate.replace("-----END CERTIFICATE-----", "")
```

Comment:
I see the reason for these replace/removal of the cert headers is to satisfy the input for the base64 check, but at the same time there is modification to the user input --- double checking we don't lose any existing compatibility here.